legacy_packages/sdk/src/evm/zksync/zksync-deploy-utils.ts

```diff
@@ -229,22 +233,70 @@ export async function zkDeployContractFromUri(
       constructorParamValues,
     );
 
-    const factory = new ZkContractFactory(
-      compilerMetadata.abi,
-      compilerMetadata.bytecode as BytesLike,
-      signer as ZkSigner,
-      "create",
-    );
-    const contract = await factory.deploy(...paramValues);
+    if (deterministicDeployment) {
+      const create2Factory = await zkDeployCreate2Factory(signer as ZkSigner);
 
-    // register on multichain registry
-    await registerContractOnMultiChainRegistry(
-      contract.address,
-      chainId,
-      compilerMetadata.fetchedMetadataUri,
-    );
+      const encodedArgs = utils.defaultAbiCoder.encode(
+        constructorParamTypes,
+        constructorParamValues,
+      );
+
+      deployedAddress = zkComputeDeploymentAddress(
+        bytecode,
+        encodedArgs,
+        create2Factory,
+        options?.saltForProxyDeploy,
+      );
+
+      if (
+        await isZkContractDeployed(deployedAddress, signer.provider as Provider)
+      ) {
+        throw new Error("Contract already deployed.");
+      }
+
+      const tx = {
+        predictedAddress: deployedAddress,
+        to: create2Factory,
+        constructorCalldata: utils.arrayify(encodedArgs),
+        bytecode,
+        bytecodeHash: utils.hexlify(hashBytecode(bytecode)),
+        abi: compilerMetadata.abi,
+        params: paramValues,
+      };
+
+      try {
+        await zkDeployContractDeterministic(
+          signer as ZkSigner,
+          tx,
+          storage,
+          compilerMetadata.fetchedMetadataUri,
+          options,
+        );
+      } catch (e) {
+        console.debug(
+          `Error deploying contract at ${tx.predictedAddress}`,
+          (e as any)?.message,
+        );
+      }
+    } else {
+      const factory = new ZkContractFactory(
+        compilerMetadata.abi,
+        compilerMetadata.bytecode as BytesLike,
+        signer as ZkSigner,
+        "create",
+      );
+      const contract = await factory.deploy(...paramValues);
+      deployedAddress = contract.address;
+    }
 
-    deployedAddress = contract.address;
+    if (deployedAddress) {
+      // register on multichain registry
+      await registerContractOnMultiChainRegistry(
```

Comment:
Do we actually need this step still?

Reply:
Yes, the ipfs hash encoding is still not available on zksolc bytecode.